### ISSUE TITLE
Filter on field added to the generate junction collection of an m2m relation does not work

### ISSUE THREAD

**Issue by magomimmo** [directus/directus#4417]
To reproduce:
1. create an Ingredient collection with a name field
2. create a Recipe collection with a name field
3. create an m2m recipe/ingredient named ingredients from the recipe collection
4. Add a quantity field to the generated recipe_ingredient collection
5. Refine the newly recipe's ingredients field by adding name and quantity
6. create few ingredient's entries
7. create few recipe's entries including few ingredients and setting the corresponding quantity

I you try to filter recipes  by ingredient, the filter does not respond.

See attached screen recording

https://user-images.githubusercontent.com/1448905/110242068-8a58fc00-7f54-11eb-9ef8-146d617d6956.mp4

---

### Project details
Directus Version: 9.0.0-rc.45
Environment: production
OS: Linux 5.4.0-66-generic
Node: 14.16.0

**Comment by magomimmo** [comment]
@benhaynes did I report something badly here? I see that it's the only one issue not classified after few days. I can close it if it seems to you unimportant. 

My best

**Comment by benhaynes** [comment]
> @benhaynes did I report something badly here? I see that it's the only one issue not classified after few days. I can close it if it seems to you unimportant.

Not at all! I think this just slipped through the cracks... you did a great job reporting it, and that video _really_ helps. We've been super busy, but will take a look ASAP. 🙂

**Comment by nordcart** [comment]
I´m having the same problem :)

**Comment by nordcart** [comment]
This now works. But filtering still needs some work?

For example: 

1. If I only want to show a list of customers which does not contain "2/2021" anywhere in **Tyot**.
2. For this example one item in **Tyot** does contain 2/2021 the rest does not. Which means it is still displayed, because the other items does not contain 2/2021.

![Screenshot_2](https://user-images.githubusercontent.com/12813484/114142281-715cb580-991b-11eb-862d-9e957ded5c1b.jpg)

**Comment by benhaynes** [comment]
My guess is that this is due to a difference between how dates are displayed in Directus (and entered into the filter input) versus the database value format. Can you check the database and see what the raw value of that field is?

**Comment by nordcart** [comment]
> My guess is that this is due to a difference between how dates are displayed in Directus (and entered into the filter input) versus the database value format. Can you check the database and see what the raw value of that field is?

I explained very confusingly.

What I mean is that I want to show only those "kundnummer" where 3/2021 is missing. 3/2021 is not a date it is a string. Is it possible to filter like that?

![image](https://user-images.githubusercontent.com/12813484/114177213-ccef6900-9944-11eb-9002-47171587b7ba.png)

**Comment by benhaynes** [comment]
If it's a **string**, you probably want to use the "Does Not Contain" or "Not Equal To" operators... not the "Does Not Include", which is used for arrays of values (eg: tags).

**Comment by nordcart** [comment]
> If it's a **string**, you probably want to use the "Does Not Contain" or "Not Equal To" operators... not the "Does Not Include", which is used for arrays of values (eg: tags).

I tried that. Did not work.

As in the above screenshot, 111 have 5 items.

If I add the filter "Does not contain 2/2021", 111 will still show up, because there are 4 other items that does not contain 2/2021.

Is there a workaround for this?
